File: FragmentPart4/app/src/main/java/com/example/fragmentpart4/MainActivity.java
package com.example.fragmentpart4;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import android.content.ClipData;
import android.content.res.Configuration;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Bundle bundle = new Bundle();
        bundle.putInt("itemPosition", 0);

        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();
        ItemFragment itemFragment = new ItemFragment();
        DetailFragment detailFragment = new DetailFragment();
        detailFragment.setArguments(bundle);
        ft.replace(R.id.itemFrame,itemFragment,"Item Fragment");
        ft.replace(R.id.detailFrame,detailFragment,"detail");
        ft.commit();

    }
}

File: GalleryAppMvvm/app/src/main/java/com/example/galleryappmvvm/view/MainActivity.kt
package com.example.galleryappmvvm.view

import android.content.Intent
import android.content.pm.PackageManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction
import androidx.lifecycle.ViewModelProvider
import com.example.galleryappmvvm.R
import com.example.galleryappmvvm.viewmodel.FirebaseViewModel

private var TAG = MainActivity::class.java.simpleName
class MainActivity : AppCompatActivity() {
    private lateinit var fragment: Fragment
    private lateinit var fragmentManager: FragmentManager
    private lateinit var fragmentTransaction: FragmentTransaction


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        addLoginFragment()
        setupPermissions()
        addLoginFragment()
        val viewModel = ViewModelProvider(this).get(FirebaseViewModel::class.java)
        val currentUser = viewModel.currentUser()
        if(currentUser != null){
            startActivity(Intent(this,CategoryActivity::class.java))
            finish()
        }
        else{
            addLoginFragment()
        }

    }

    private fun setupPermissions() {
        val permission = ContextCompat.checkSelfPermission(this, android.Manifest.permission.CAMERA)
        if (permission != PackageManager.PERMISSION_GRANTED) {
            Log.d(TAG, "Permission to camera denied")
            makePermissionRequest()
        }
    }

    private val CAMERA_REQUEST_CODE = 101
    private fun makePermissionRequest() {
        ActivityCompat.requestPermissions(
            this,
            arrayOf(android.Manifest.permission.CAMERA),
            CAMERA_REQUEST_CODE
        )
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<out String>,
        grantResults: IntArray
    ) {
        when (requestCode) {
            CAMERA_REQUEST_CODE -> {
                if (grantResults.isEmpty() || grantResults[0] != PackageManager.PERMISSION_GRANTED) {
                    Log.d(TAG, "Permission has been denied by user")
                } else {
                    Log.d(TAG, "Permission has been granted by user")
                }
            }
        }
    }

    private fun addLoginFragment() {
        fragment = LoginFragment()
        fragmentManager = supportFragmentManager
        fragmentTransaction = fragmentManager.beginTransaction()
        fragmentTransaction.add(R.id.fragment_container, fragment)
        fragmentTransaction.commit()
    }

}
